bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -849,7 +839,7 @@ PrePrepareMsgCreationResult ReplicaImp::buildPrePrepareMessageByRequestsNum(uint
   }
 
   uint32_t maxSpaceForReqs = prePrepareMsg->remainingSizeForRequests();
-  ClientRequestMsg *nextRequest = requestsQueueOfPrimary.front();
+  ClientRequestMsg *nextRequest = requestsQueueOfPrimary.front().get();
```

Comment:
any idea why here and in  line  859   and further down we do not check requestsQueueOfPrimary.empty() line in line 825 for example?

Reply:
Maybe the logic assumes that it does not happen. For safety I  will introduce the empty check.

Reply:
Done